firefox-ios/Client/Frontend/Home/Homepage Rebuild/HomepageViewController.swift

```diff
@@ -193,41 +185,44 @@ final class HomepageViewController: UIViewController,
             return UICollectionViewCell()
         }
 
-        cell.contentView.subviews.forEach { $0.removeFromSuperview() }
-        cell.contentView.backgroundColor = getBackgroundColor(for: section)
-
-        let label = UILabel(frame: cell.contentView.bounds)
-        label.textAlignment = .center
-        label.text = item.title
-        label.textColor = .white
-        cell.contentView.addSubview(label)
-
-        return cell
-    }
-
-    private func getBackgroundColor(for section: HomepageSection) -> UIColor {
         switch section {
         case .header:
-            return .systemPink
-        case .topSites:
-            return .systemGreen
-        case .pocket:
-            return .systemPurple
-        }
-    }
-
-    // MARK: UICollectionViewDelegate
-    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        if let selectedItem = dataSource?.itemIdentifier(for: indexPath) {
-            // TODO: FXIOS-10162 - Dummy trigger to update with proper triggers
-            if selectedItem.title == "First" {
+            let cell = collectionView?.dequeueReusableCell(
+                withReuseIdentifier: HomepageHeaderCell.cellIdentifier,
+                for: indexPath
+            )
+            guard let headerCell = cell as? HomepageHeaderCell else {
+                return UICollectionViewCell()
+            }
+            headerCell.configure(
+                headerState: homepageState.headerState,
```

Comment:
I was a bit torn if this state should come from our updated `newState` but I decided to opt into taking the simpler path since this state doesn't really change after initialization. However, in some of the other section cases, I will need to pass in the items to the snapshot and so those states will come from the updated `newState`. 

@OrlaM maybe I can grab some time on your calendar when you feel better to chat on this.